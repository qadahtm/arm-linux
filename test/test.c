#include <unistd.h> 
#include <stdio.h>

typedef struct pte_refc  pte_refc_t;
struct pte_refc {
   int refcount;
   //pte_t *pte;
   int *pte;
   pte_refc_t *next_pte;
};

typedef struct proc_refc proc_refc_t;

struct proc_refc
{
    //pid_t pid;	
   int pid; 
   pte_refc_t *pte_list;
   proc_refc_t *next;
};

int main(){ 

//sleep(10000); 
printf("here\n");

proc_refc_t * ref_list = (proc_refc_t *) malloc(sizeof(proc_refc_t));
proc_refc_t *proc = NULL;

ref_list->pid = 0;
ref_list->pte_list = (pte_refc_t *) malloc(sizeof(pte_refc_t));
ref_list->next = NULL;

proc = (proc_refc_t *) malloc(sizeof(proc_refc_t));
proc->pid = 1;
proc->pte_list = (pte_refc_t *) malloc(sizeof(pte_refc_t));
proc->next = NULL;

ref_list->next = proc;

proc = (proc_refc_t *) malloc(sizeof(proc_refc_t));
proc->pid = 2;
proc->pte_list = (pte_refc_t *) malloc(sizeof(pte_refc_t));
proc->next = NULL;

ref_list->next->next = proc;

proc = ref_list;
while (proc != NULL) {
   printf("pid = %d\n", proc->pid);
   proc = proc->next;
}

return 0;

}
