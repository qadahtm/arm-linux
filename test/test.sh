./cpu_task & 
./io_task &
