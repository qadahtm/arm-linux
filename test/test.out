sleeping for 3 secs
sleeping for 1 sec
read again: faulty pc 000086cc, instr e7f001f2
patch is done
hpte was 9fff9e7c. going to enable it.
xzltestprog (1421): undefined instruction: pc=000086cc
Code: e3a00001 ebffff5d e51b3008 e5933000 (e7f001f2) 
oh this undefinstr is planted by us. 
read: faulty pc 000086cc, instr e7f001f2
restore instr ok -- user good to go
read again: faulty pc 000086cc, instr e50b3014
hpte was 9fff9e7f, going to disable it.
sleeping for 1 sec
mmap p 40000000
will read in 1 sec...
read once. 0
will read in 1 sec...
read once. 1
will read in 1 sec...
read once. 2
will read in 1 sec...
read once. 3
will read in 1 sec...
read once. 4
will read in 1 sec...
read once. 5
will read in 1 sec...
read once. 6
will read in 1 sec...
read once. 7
will read in 1 sec...
read once. 8
will read in 1 sec...
read once. 9
bye!
